buildpacks/java/java/detect.go

```diff
@@ -9,6 +9,8 @@ import (
 	"github.com/buildpacks/libcnb"
 	"github.com/paketo-buildpacks/libpak"
 	"github.com/paketo-buildpacks/libpak/bard"
+
+	"kn-fn/shared/shared"
```

Comment:
🤔 Does this import path seem good as a structure?

Reply:
I wondered the same. I was following what I saw in our other go subprojects in this repo, which was to have a structure like:

```
<project dir>/
    <primary package>/
    tests/
```

where the main code lives under `<primary package>/`. I originally had it where the main code was up one level, and the import looked a little nicer (`kn-fn/shared`). It's just aesthetics though, so I prioritized what I saw as our convention.

Happy to go either way though. No preference from me.

Reply:
@bbtong @andrew-su @joeeltgroth any preferences here? Happy to follow whatever the consensus is.

Reply:
Another option is to name the module itself something other than `kn-fn/shared`, then keep the current folder structure. For instance, if the module were named `kn-fn/buildpacks`, the import could be `kn-fn/buildpacks/shared`. Again, it's mostly aesthetics though, so not sure how much we care.

Reply:
`kn-fn/shared/shared` is definitely a bit awkward to say. I'm okay with either being just `kn-fn/shared` or `kn-fn/buildpacks/shared` with preference for the latter. It's more explicit that it's for the buildpacks (though we don't currently use Go elsewhere)

Reply:
Will change module name to `kn-fn/buildpacks` per this thread: https://github.com/vmware-tanzu/function-buildpacks-for-knative/pull/137/files#r987176488

Reply:
Removed shared code for this PR and will address in later PR